src/build.h

```diff
@@ -233,6 +233,11 @@ struct Builder {
   std::unique_ptr<CommandRunner> command_runner_;
   Status* status_;
 
+  /// Keep the global exit code for the run
+  ExitStatus exit_code_;
```

Comment:
nit: move this variable and the `SetExitCode()` function to the `private` section to avoid direct access. Maybe also rename `ExitCode()` to `GetExitCode()` for clarity.